src/auth/email/email.service.ts

```diff
@@ -0,0 +1,186 @@
+import { Inject, Injectable, Logger } from '@nestjs/common';
+import { LoginDto } from './dto/login.dto';
+import { LoginResponseDto } from './dto/login-response.dto';
+import { AuthTokenService } from '../core/auth-token/auth-token.service';
+import { AuthTypeEnum, User } from '@prisma/client';
+import { RequestException } from 'src/common/exception/core/ExceptionBase';
+import { Exceptions } from 'src/common/exception/exceptions';
+import { RegisterDto } from './dto/register.dto';
+import forgotPasswordEmail from '../../email/emails/forgot-password.email';
+import EmailDeliveryIntegratorFactory, {
+  EmailDeliveryIntegratorName,
+} from 'src/common/factory/emailDeliveryIntegrator/emailDeliveryIntegrator.factory';
+import { EmailType } from 'src/email/core/email-type';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { EncryptService } from '../core/auth-token/encrypt.service';
+
+@Injectable()
+export class EmailAuthService {
+  private readonly logger = new Logger(this.constructor.name, {
+    timestamp: true,
+  });
+
+  constructor(
+    private authTokenService: AuthTokenService,
+    private prisma: PrismaService,
+    private encryptService: EncryptService,
+    @Inject(forgotPasswordEmail.KEY)
+    private readonly forgotPasswordMail: EmailType<typeof forgotPasswordEmail>,
+  ) {}
+
+  async login(login: LoginDto): Promise<LoginResponseDto> {
+    try {
+      const user = await this.prisma.user.findUnique({
+        email: login.email,
+        deleted: false,
+      });
+      if (
+        !user ||
+        !(await this.encryptService.validPassword(
```

Comment:
uh uh, can we try to move these complex expressions in the conditional out of the conditional? This makes the code more readable!

```ts
const isPasswordValid = await this.encryptService.validPassword(...)

if (!user || !isPasswordValid) { ...
```